src/rebar3_hex_docs.erl

```diff
@@ -156,6 +157,32 @@ resolve_doc_dir(AppInfo) ->
     Dir = proplists:get_value(dir, EdocOpts, ?DEFAULT_DOC_DIR),
     proplists:get_value(doc, AppDetails, Dir).
 
+maybe_gen_docs(State) -> 
+    case doc_opts(State) of
+        undefined -> 
+             rebar_api:error("No docs config found", []),
+            noop;
+        PrvName -> 
+            AllProviders = rebar_state:providers(State),
+             case providers:get_provider(PrvName, AllProviders) of
+                 not_found -> 
+                     rebar_api:error("No provider found for ~ts", [PrvName]),
+                     noop;
+                 Prv ->
+                     providers:do(Prv, State)
+             end
+    end.
+
+doc_opts(State) -> 
+    Opts = rebar_state:opts(State),
+    case proplists:get_value(doc, rebar_opts:get(Opts, hex), undefined) of
+        PrvName when is_atom(PrvName) -> PrvName;
```

Comment:
should wrap in `{ok, PrvName}` to avoid the case of `undefined` being a module that doesn't work :P